app/resources/services/integrations/patreon.js

```diff
@@ -0,0 +1,120 @@
+const config = require('config')
+const logger = require('../../../../lib/logger.js')()
+
+const { User } = require('../../../db/models')
+const passport = require('passport')
+const PatreonStrategy = require('passport-patreon').Strategy
+
+/*
+our use case makes this a little complicated,
+we basically have three user instances to check:
+
+ * Auth0 data on a user who is authenticated (aka 'user)
+ * Data on this user that we store in our database (aka 'databaseUser')
+ * Data from a user's third party account (aka profile)
+
+Our existing code handles login using the auth0 service,
+which if valid we store a JWT.
+
+We pass this to the initial get route,
+which passes along the account id/name (and errors if no one is signed in)
+
+We then pass this to our passport strategy, which grabs third party data and
+looks up the user in _our_ database, which then gets handled by the callback function
+an subsequently links all this info together
+
+*/
+
+/**
+ finds the database record for the given user
+ */
+const findUser = async function (userId) {
+  const user = await User.findByPk(userId)
+  if (user === null) {
+    return 'user not found'
+  }
+  return user.dataValues
+}
+
+passport.use(
+  new PatreonStrategy(
+    {
+      clientID: process.env.PATREON_CLIENT_ID,
+      clientSecret: process.env.PATREON_CLIENT_SECRET,
+      callbackURL: `${config.restapi.protocol}${config.app_host_port}/services/integrations/patreon/callback`,
+      scope: 'users pledges-to-me my-campaign',
+      passReqToCallback: true
+    },
+    async function (req, accessToken, refreshToken, profile, done) {
+      /*
+      this is what passport calls the 'verify callback'
+      our case is a little different since we already have a logged in user
+      not sure if this is the best place to handle checking the DB or not but it works
+       */
+      const databaseUser = await findUser(req.query.state)
+      // passing the profile data along the request, probably another way to do this
+      req.profile = profile
+      return done(null, databaseUser)
+    }
+  )
+)
+
+// these aren't used yet, but would be if we start using persistent user sessions
+passport.serializeUser(function (user, done) {
```

Comment:
since we don't use these yet, I could take these out? We can talk more about what I think they might eventually be used for

Reply:
Yeah - let's talk more about this, but I'll leave this in for now to facilitate merging this soon